ledger-tool/src/main.rs

```diff
@@ -2074,13 +2104,33 @@ fn main() {
                 );
                 exit(1);
             }
-            println!("Purging data from slots {} to {}", start_slot, end_slot);
-            if no_compaction {
-                blockstore.purge_slots(start_slot, end_slot, PurgeType::Exact);
-            } else {
-                blockstore.purge_and_compact_slots(start_slot, end_slot);
+            info!(
+                "Purging data from slots {} to {} ({} slots) (skip compaction: {})",
+                start_slot,
+                end_slot,
+                end_slot - start_slot,
+                no_compaction,
+            );
+            for slots in &(start_slot..=end_slot).chunks(batch_size) {
```

Comment:
Hmm, with dead_slot_iterator, we can do better by implementing `ledger-tool purge --only-dead-slots`? Or, too dangerous to add it at this moment? ;)

Reply:
Hmm, I think we should lean to the safer side. Still, the support incident's root cause is unclear. Purging only dead slots might not be enough...

Reply:
>  I think we should lean to the safer side. Still, the support incident's root cause is unclear. Purging only dead slots might not be enough...

On the other hand, I'm starting to suspect a validator can handle the repair_state which will contain around 2_200_000 worth of slots to repair.... So, maybe deleting dead slots might be the only way forward?

@carllin do you have  rough estimate for the required ram? Also, maybe flushing repairs to disk isn't small work, right?

Reply:
Cool, solana-validator even can repair from snapshot-39258079-7o86jrh3NJNjYxx4pWcPa7fSwAZvX4HynXdhynZ98VF2.tar.zst, it seems atm.

So there is no problem.

We want to resume from around slot 41040067

Reply:
> Hmm, with dead_slot_iterator, we can do better by implementing `ledger-tool purge --only-dead-slots`?

I just added this in case for the backup plan: https://github.com/solana-labs/solana/pull/13065/commits/61c38ac855d818f74d2b5c52299c6a81c637b0bc